### PR TITLE
Add Phone Home capability and refactor code

### PR DESCRIPTION
**Pull request by jasonf1234**
**Description:**
Added a new job to send usage data to Black Duck. This job runs on the
same schedule as the push scan data task. In the process refactored code
so that PropertyConstants and AttributeConstants were no longer
statically accessible (accessing the static variables when the Spring
bean was being instantiated caused null return values and consequently
validation method were failing).

* Edited application.properties and application_prod.properties to have
a placeholder for the plugin version, and edited the build.gradle to
fill in that placeholder using values from the project.properties.
* Ensure PropertyConstants and AttributeConstants are loaded as Spring
Beans
* Removed `static` keyword from all PropertyConstants and
AttributeConstants fields and methods.
* Edited all locations where PropertyConstants and/or AttributeConstants
were used to either provide a Dependency Injected bean or pass the bean
through the constructor for non- spring bean classes
* Added new Job Configuration and Step classes for Phone Home
capability. Job runs on the same schedule as the push scan data task
* Modified all JUnit tests to use Dependency Injected beans where
possible

**Reviewers:**
@robpacheco 
@skatzman 
@msenmurugan

### CODE REVIEW

File: src/main/java/com/blackducksoftware/integration/fortify/service/FortifyAttributeDefinitionApi.java [hunk 1]
```diff
@@ -40,13 +40,20 @@
 
     private final static String FIELDS_ATTRIBUTE = "id,name,category,type,options,required";
 
-    private final static OkHttpClient.Builder okBuilder = getHeader(PropertyConstants.getFortifyUserName(),
-            PropertyConstants.getFortifyPassword());
+    private static OkHttpClient.Builder okBuilder;
```
Comment (msenmurugan) [inline src/main/java/com/blackducksoftware/integration/fortify/service/FortifyAttributeDefinitionApi.java]: Can you please remove static and add final. Similar comments for other fields as well

Comment (jasonf1234) [reply src/main/java/com/blackducksoftware/integration/fortify/service/FortifyAttributeDefinitionApi.java]: Made the changes we discussed. Could you please review again and approve.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private PropertyConstants propertyConstants;
+
+    /**
+     * Constructor
+     *
+     * @param hubServices
+     *            HubServices object which provides access to the Hub
+     */
+    public BlackDuckFortifyPhoneHomeStep(HubServices hubServices, PropertyConstants propertyConstants) {
+        super();
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: I don't think we need this

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * @author jfisher
+ *
+ */
+public class BlackDuckFortifyPhoneHomeStep implements Tasklet, StepExecutionListener {
+    private final static Logger logger = Logger.getLogger(BlackDuckFortifyPhoneHomeStep.class);
+
+    private String startJobTimeStamp;
+
+    private HubServices hubServices;
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: Should these two be final?

Comment (jasonf1234) [reply src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: Actually, wouldn't I want to make all three (startJobTimeStamp, hubServices, and propertyConstants) final?

File: src/test/java/com/blackducksoftware/integration/fortify/service/FortifyFileTokenApiTest.java [hunk 4]
```diff
@@ -23,43 +23,44 @@
...
 
-    @Override
-    @Before
-    public void setUp() {
-        blackDuckFortifyJobConfig = new BlackDuckFortifyJobConfig();
-    }
+    @Autowired
+    private FortifyFileTokenApi fortifyFileTokenApi;
 
     @Test
```
Comment (skatzman) [inline src/test/java/com/blackducksoftware/integration/fortify/service/FortifyFileTokenApiTest.java]: I think the build system handles it appropriately, but we've standardized on using TestNG over JUnit due some of the features it provided over JUnit.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java (hunk 2 above)
Comment (skatzman) [reply src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: +1

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java (hunk 3 above)
Comment (skatzman) [reply src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: Ideally, yes.

File: src/test/java/com/blackducksoftware/integration/fortify/service/FortifyFileTokenApiTest.java (hunk 4 above)
Comment (jasonf1234) [reply src/test/java/com/blackducksoftware/integration/fortify/service/FortifyFileTokenApiTest.java]: I will log an issue to move the tests to use the TestNG framework.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java (hunk 2 above)
Comment (jasonf1234) [reply src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: Issue remediated.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java (hunk 3 above)
Comment (jasonf1234) [reply src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java]: I was able to make `hubServices` and `propertyConstants` `final`. I could not make `startJobTimeStamp` as that is updated by the overridden `beforeStep` method.

**Pull request merged**